### PR TITLE
tests: new set of nested tests for uc22

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes:
 . Updates for the tests to support uc22
 . New models for uc22
 . New snap needed to run uc22 tests

### CODE REVIEW

File: tests/lib/gendeveloper1model/main.go [hunk 1]
```diff
@@ -58,7 +57,13 @@ func main() {
 		log.Fatalf("failed to decode model headers data: %v", err)
 	}
 
-	clModel, err := devSigning.Sign(asserts.ModelType, headers, nil, "")
+	typCand, _ := headers["type"]
+	typ := asserts.ModelType
```
Comment (mardy) [inline tests/lib/gendeveloper1model/main.go:61]: nitpick: please don't abbreviate variable names, unless the abbreviation is commonly used already ("dev" for device is fine, but I have no idea what "cand" is, and if "typ" stands for "type", I find it more readable if it's written in full :-) )

File: tests/lib/snaps/test-snapd-tools-core22/bin/sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+Launching a shell inside the default app confinement. Navigate to your
+app-specific directories with:
+
+  $ cd \$SNAP
+  $ cd \$SNAP_DATA
+  $ cd \$SNAP_USER_DATA
+
+EOM
+
+/bin/bash --norc -i
```
Comment (mardy) [inline tests/lib/snaps/test-snapd-tools-core22/bin/sh:13]: Could this file (and many others) be a symbolic link?

**Review by mardy — APPROVED** [review]
LGTM, but I wonder if some of those added files can by symlinks instead.

File: tests/lib/gendeveloper1model/main.go (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/gendeveloper1model/main.go:61]: @mardy Thanks for the review, I'll update this

File: tests/lib/snaps/test-snapd-tools-core22/bin/sh (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-snapd-tools-core22/bin/sh:13]: This is the same code that we use for the other test-snapd-tools snaps, so I would not change that once in this pr.
Also the snap is already published.
We could make that change in another pr and all the snaps should be updated in the store to be consistent

Comment (mardy) [reply tests/lib/snaps/test-snapd-tools-core22/bin/sh:13]: OK!

**Review by mardy — APPROVED** [review]
Thank for the update, still LGTM :-)

**Pull request merged**